docs/SSH_KEY_RETENTION.md

```diff
@@ -0,0 +1,199 @@
+# SSH Key Retention for Git Proxy
```

Comment:
Nit: Replace all instances of "Git Proxy" with "GitProxy" without a space for consistency 👍🏼